opsdroid/connector/slack/events.py

```diff
@@ -42,3 +42,39 @@ def __init__(self, blocks, *args, **kwargs):
         self.blocks = blocks
         if isinstance(self.blocks, list):
             self.blocks = json.dumps(self.blocks)
+
+
+class BlockActions(Event):
+    """Event class to represent block_actions in Slack."""
+
+    def __init__(self, payload, *args, **kwargs):
+        """Create object with minimum properties."""
+        super().__init__(*args, **kwargs)
+        self.payload = payload
+
+
+class MessageAction(Event):
+    """Event class to represent message_action in Slack."""
+
+    def __init__(self, payload, *args, **kwargs):
+        """Create object with minimum properties."""
+        super().__init__(*args, **kwargs)
+        self.payload = payload
+
+
+class ViewSubmission(Event):
+    """Event class to represent view_submission in Slack."""
+
+    def __init__(self, payload, *args, **kwargs):
+        """Create object with minimum properties."""
+        super().__init__(*args, **kwargs)
+        self.payload = payload
+
+
+class ViewClosed(Event):
+    """Event class to represent view_closed in Slack."""
+
+    def __init__(self, payload, *args, **kwargs):
+        """Create object with minimum properties."""
+        super().__init__(*args, **kwargs)
+        self.payload = payload
```

Comment:
As all these classes are the same other than the docstrings you could create a superclass with this `__init__` method and inherit all classes from that.

Reply:
Also many of these actions are given a `response_url` which should be used when responding to the action. We may need to handle this in the connector.

```python
from opsdroid.skill import Skill
from opsdroid.matchers import match_event
from opsdroid.connector.slack.events import MessageAction

class InteractionsSkill(Skill):
    @match_event(MessageAction)
    async def slack_interactions(self, event):
        await event.respond("Message Action interactivity has been triggered.")  # What happens here?
```

Reply:
Superclass added with `respond` method.